### PR TITLE
Use "blur_sigma" instead of "blur_radius" in Shadow.

### PR DESCRIPTION
**Pull request by heke123**
This is a bug-fix. (https://github.com/flutter/flutter/issues/81145)
issue: #81145

The Skia 'SkMaskFilter::MakeBlur()' interface expects a 'blur_sigma'
value but currently we pass it a 'radius'.

The skt::TextShadow class also direcly passes its |fBlurRadius|
directly to SkMaskFilter::MakeBlur() without converting it to a
'blur_sigma', which makes the |fBlurRadius| member actually means
"blur_sigma" instead of "blur_radius".

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [x] I listed at least one issue that this PR fixes in the description above.
- [ ] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt. See [testing the engine] for instructions on
writing and running engine tests.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I signed the [CLA].
- [ ] All existing and new tests are passing.
- [ ] The reviewer has submitted any presubmit flakes in this PR using the [engine presubmit flakes form] before re-triggering the failure.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/master/CONTRIBUTING.md#style
[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[engine presubmit flakes form]: https://forms.gle/Wc1VyFRYJjQTH6w5A

### CODE REVIEW

**Comment by flar** [comment]
This PR is modifying some of the code in the imported third party libraries so I think @jason-simmons should be looking at this and considering the import status of the code in whether these changes will make tracking the third party code harder in the long run.

Also, at least one of the data members still uses the old "radius" naming even though it is holding a "sigma" value. Should it be changed to Sigma as well?

**Comment by jason-simmons** [comment]
LGTM

The code in `flutter/third_party` originated from external sources, but it is effectively now a fork and is not actively kept in sync with the upstream.

However, the SkParagraph `TextShadow.fBlurRadius` field is defined in code that lives in Skia's repository.  It looks like SkParagraph copied Libtxt's use of the "blur radius" name and is also passing that value to `SkMaskFilter::MakeBlur` which is expecting a sigma value.
(https://github.com/google/skia/blob/master/modules/skparagraph/src/TextLine.cpp#L368)

SkParagraph should be updated similarly to this PR.  But that will need to happen separately in the Skia sources.

**Comment by flar** [comment]
Would it make sense to file a Skia bug about that and then a "follow-on" Flutter bug to fix the naming in this code when that gets addressed?

**Comment by jason-simmons** [comment]
Sent a PR to update SkParagraph: https://skia-review.googlesource.com/c/skia/+/401496

**Comment by jason-simmons** [comment]
The change to update SkParagraph and the engine to use `TextShadow.fBlurSigma` instead of `fBlurRadius` has landed.

This PR will need to be rebased on top of the latest engine.

File: third_party/txt/src/txt/text_shadow.cc [hunk 1]
```diff
@@ -41,7 +41,7 @@ bool TextShadow::operator!=(const TextShadow& other) const {
 bool TextShadow::hasShadow() const {
   if (!offset.isZero())
     return true;
-  if (blur_radius != 0.0)
+  if (blur_sigma != 0.0)
```
Comment (flar) [inline third_party/txt/src/txt/text_shadow.cc:44]: We have an issue here. A 0 radius does not convert into a 0 sigma, so the test for 0 no longer works.

One thing I notice is that the dart conversion function used above says that it is based on the Skia version and provides a link to the Skia conversion code. If you take a look at the Skia equations, they make sure that a radius of 0 produces a sigma of 0 and that a sigma of less than 0.5 produces a radius of 0.

The net effect should be that we might want to update our Dart conversion function to match the 0 => 0 behavior of the Skia function and to change this test to `blur_sigma <= 0.5`?

**Review by flar — CHANGES_REQUESTED** [review]

File: third_party/txt/src/txt/text_shadow.cc (hunk 1 above)
Comment (heke123) [reply third_party/txt/src/txt/text_shadow.cc:44]: make sence, thanks!   the test should be "blur_sigma > 0.5" which means it is a valid value. right?

**Comment by heke123** [comment]
@flar  I didn't figure out why the "Linux Android AOT Engine" failed.  Seems I don't have the privilege to re-trigger it, could you please give a help?   Thanks.

File: third_party/txt/src/txt/text_shadow.cc (hunk 1 above)
Comment (flar) [reply third_party/txt/src/txt/text_shadow.cc]: Right, `> 0.5` means it will do something. I got that backwards.

**Review by flar — APPROVED** [review]
LGTM

**Comment by flar** [comment]
The test failure can be resolve by rebasing the fix to the top of tree.

**Comment by flar** [comment]
This will merge when the tree goes green. Thanks!

**Pull request merged**